File: backend/app/schemas/report.py
from pydantic import BaseModel


class TripKpiSummary(BaseModel):
    total_trips: int
    on_time_rate: float | None = None
    avg_delay_minutes: float | None = None
    avg_profit_margin: float | None = None
    active_trips: int
    delayed_trips: int


class StatusBucket(BaseModel):
    status: str
    count: int


class DelayBucket(BaseModel):
    label: str  # e.g. "On time", "≤30m", "31–60m", "61–90m", ">90m"
    count: int


class TripKpiDetail(TripKpiSummary):
    status_distribution: list[StatusBucket] = []
    delay_buckets: list[DelayBucket] = []

File: frontend/src/types/report.ts
export interface TripKpiSummary {
  total_trips: number;
  on_time_rate: number | null;
  avg_delay_minutes: number | null;
  avg_profit_margin: number | null;
  active_trips: number;
  delayed_trips: number;
}

export interface StatusBucket {
  status: string;
  count: number;
}

export interface DelayBucket {
  label: string;
  count: number;
}

export interface TripKpiDetail extends TripKpiSummary {
  status_distribution: StatusBucket[];
  delay_buckets: DelayBucket[];
}
